skyportal/models.py

```diff
@@ -3515,6 +3521,312 @@ class SourceNotification(Base):
     level = sa.Column(sa.String(), nullable=False)
 
 
+class Event(Base):
+    """Event information, including an event ID, mission, and time of the event."""
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+
+    dateobs = sa.Column(sa.DateTime, doc='Event time', unique=True, nullable=False)
+
+    gcn_notices = relationship(lambda: GcnNotice, order_by=lambda: GcnNotice.date)
+
+    _tags = relationship(
+        lambda: Tag,
+        order_by=lambda: (
+            sa.func.lower(Tag.text).notin_({'fermi', 'swift', 'amon', 'lvc'}),
+            sa.func.lower(Tag.text).notin_({'long', 'short'}),
+            sa.func.lower(Tag.text).notin_({'grb', 'gw', 'transient'}),
+        ),
+    )
+
+    localizations = relationship(lambda: Localization)
+
+    @property
+    def tags(self):
+        """List of tags."""
+        return [tag.text for tag in self._tags]
+
+    @hybrid_property
+    def retracted(self):
+        """Check if event is retracted."""
+        return 'retracted' in self.tags
+
+    @retracted.expression
+    def retracted(cls):
+        """Check if event is retracted."""
+        return sa.literal('retracted').in_(cls.tags)
+
+    @property
+    def lightcurve(self):
+        """GRB lightcurve URL."""
+        try:
+            notice = self.gcn_notices[0]
+        except IndexError:
+            return None
+        root = lxml.etree.fromstring(notice.content)
+        elem = root.find(".//Param[@name='LightCurve_URL']")
+        if elem is None:
+            return None
+        else:
+            return elem.attrib.get('value', '').replace('http://', 'https://')
```

Comment:
this is probably pretty brittle, especially without schema validation (or is that done upstream)? If there's any chance that this will break, you should do another try/except here.

Reply:
We have used this with success the last while, but no problem being extra careful. Are you thinking something like

```
return elem.attrib.get('value', '').replace('http://', 'https://')
->
try:
    return elem.attrib.get('value', '').replace('http://', 'https://')
else:
    return None
```

Reply:
Yes, something like this would be more robust.